### PR TITLE
Overhauled presentation in lecture 4

### PR DESCRIPTION
**Pull request by AlinderS**
Reworked the presentation. Now it follows a single function and the tools are used to improve it in different ways. 

I see there is a problem with merging, I will fix.

### CODE REVIEW

**Comment by AlinderS** [comment]
I think I did it right and we are good to go.
I also deleted xy.txt as it is not used any more.

**Comment by AlinderS** [comment]
I looked through the manual and everything that was added in the presentation was already there. I did not see a need for it to be updated.

File: 4-tools/spyderexample.py [hunk 1]
```diff
@@ -1,28 +1,19 @@
 #!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 """
-Created on Mon Nov  9 18:03:24 2020
+Created on Thu Nov  4 15:39:32 2021
 
-@author: danielmikkola
+@author: Simon
 """
```
Comment (eerovaher) [inline 4-tools/spyderexample.py:7]: None of this is useful.

File: 4-tools/spyderexample.py [hunk 2]
```diff
@@ -1,28 +1,19 @@
...
-            ydata_new[i] = np.mean(ydata[(i-k):])
-            xdata_new[i] = np.mean(xdata[(i-k):])
-        else:
-            ydata_new[i] = np.mean(ydata[(i-k):(i+k)])
-            xdata_new[i] = np.mean(xdata[(i-k):(i+k)])
-    return(xdata_new, ydata_new)
+def sieve(n):
+    primes=[]
+    test=list(range(3,n+1, 2) )
+    while test[0]<=n**0.5:
```
Comment (eerovaher) [inline 4-tools/spyderexample.py:12]: Why is this not PEP 8 compliant?

**Review by eerovaher — CHANGES_REQUESTED** [review]

File: 4-tools/spyderexample.py (hunk 2 above)
Comment (AlinderS) [reply 4-tools/spyderexample.py:12]: In order to demonstrate pycodestyle_magic on it later in the presentation.

Comment (eerovaher) [reply 4-tools/spyderexample.py:12]: The file should be renamed then, it is not obvious that it will be used for anything other than demonstrating Spyder.

Comment (AlinderS) [reply 4-tools/spyderexample.py]: Done

File: 4-tools/spyderexample.py (hunk 1 above)
Comment (AlinderS) [reply 4-tools/spyderexample.py]: Sure. Spyder adds it automatically but I removed it now.

**Pull request merged**